packages/mermaid/src/rendering-util/rendering-elements/nodes.ts

```diff
@@ -1,10 +1,21 @@
 import { log } from '../../logger.js';
 import { shapes } from './shapes.js';
+import type { Node } from '../types.js';
+import type { MermaidConfig, SVGGroup } from '../../mermaid.js';
+import type { D3Selection } from '../../types.js';
+import type { graphlib } from 'dagre-d3-es';
 
-const nodeElems = new Map();
+type ShapeHandler = (typeof shapes)[keyof typeof shapes];
+type NodeElement = D3Selection<SVGAElement> | Awaited<ReturnType<ShapeHandler>>;
 
-export const insertNode = async (elem, node, renderOptions) => {
-  let newEl;
+const nodeElems = new Map<string, NodeElement>();
+
+export async function insertNode(
+  elem: SVGGroup,
+  node: Node,
+  renderOptions: { config: MermaidConfig; dir: Node['dir'] }
```

Comment:
```suggestion
  renderOptions: ShapeRenderOptions
```

Reply:
Good catch! I've fixed this in https://github.com/mermaid-js/mermaid/pull/5974/commits/4bc70b732560102d481110f4344b5a4c6776c335